selenium_tests/course_info/course_info_tests.py

```diff
@@ -0,0 +1,106 @@
+import urlparse
+
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+
+from selenium_tests.course_info.course_info_base_test_case import CourseInfoBaseTestCase
+from selenium_tests.course_info.page_objects.editor_page import EditorPage
+from selenium_tests.course_info.page_objects.main_page import MainPage
+from selenium_tests.pin.page_objects.pin_login_page_object import PinLoginPageObject
+
+
+class CourseInfoTestFlow(CourseInfoBaseTestCase):
+    def test_course_info_tool_loads(self):
+        """
+        test that the course info tool loads properly.
+        """
+        # try to load the base url, but expect to be interrupted by the pin
+        # login page
+        pin_page = PinLoginPageObject(self.driver)
+        pin_page.login(self.BASE_URL, self.USERNAME, self.PASSWORD)
+        edit_page = EditorPage(self.driver)
+        self.assertTrue(edit_page.is_loaded(), 'edit page not loaded')
+        edit_page.focus_on_default_content()
+
+        # click on our tool's button in the editor, verify the tool's field
+        # selection page comes up in the "modal" iframe
+        edit_page.tool_button.click()
+        edit_page.focus_on_tool_frame()
+        WebDriverWait(self.driver, 30).until(
+            lambda d: edit_page.registrar_code_checkbox.is_displayed(),
+            'Timed out waiting for registrar code checkbox to be displayed')
+
+        # just for completeness sake, let's verify that *all* checkboxes are there
+        checkbox_properties = [p for p in edit_page.located_properties
+                                   if p.endswith('_checkbox')]
+        for checkbox in checkbox_properties:
+            self.assertTrue(getattr(edit_page, checkbox).is_displayed(),
+                            '{} is not displayed'.format(checkbox))
+
+
+    def test_insert_widget_all_fields(self):
+        """
+        Tests inserting the widget into a page, leaving all fields selected.
+        """
+        # deal with the login
+        pin_page = PinLoginPageObject(self.driver)
+        pin_page.login(self.BASE_URL, self.USERNAME, self.PASSWORD)
+        edit_page = EditorPage(self.driver)
+
+        # delete any instances of the widget already in the page
+        edit_page.focus_on_default_content()
+        edit_page.focus_on_editor_frame()
+        for widget in edit_page.get_inserted_widgets():
+            widget.click()
+            edit_page.editor_body.send_keys(Keys.DELETE)
+
+        # now squawk if any are left
+        edit_page.focus_on_default_content()
+        edit_page.focus_on_editor_frame()
+        self.assertEqual(edit_page.get_inserted_widgets(), [],
+                         'Found unexpected instances of widget already in the page')
+
+        # launch the tool, click save in the tool iframe
```

Comment:
I don't think we should be moving any verification/assert logic into the page objects.  But just like the other ones, I definitely agree this "do some focus logic and X" chunk belongs in a method.